src/snowflake/snowpark/mock/_nop_analyzer.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
```

Comment:
Should we remove as part of this PR (or maybe followup) code like the following which has been used so far to make some tests pass? This here looks like a MUCH BETTER solution, so I'd prefer to get rid off the old code.

```
 if (
            isinstance(self._session._conn, MockServerConnection)
            and self._session._conn._suppress_not_implemented_error
        ):
            # Allow AST tests to pass.
            return []
```

links for examples:
- https://github.com/snowflakedb/snowpark-python/blob/ls-SNOW-1491199-merge-phase0-server-side/src/snowflake/snowpark/dataframe.py#L1834
- https://github.com/snowflakedb/snowpark-python/blob/ls-SNOW-1491199-merge-phase0-server-side/src/snowflake/snowpark/dataframe.py#L4009

Ideally we would remove from session the `_suppress_not_implemented_error` attribute.

Reply:
Created https://snowflakecomputing.atlassian.net/browse/SNOW-1782942 to follow up on this and other clean up of dead code related to emit ast processing in local test mode.